test/cpp/lazy/test_backend_device.cpp

```diff
@@ -86,8 +89,8 @@ TEST(BackendDeviceTest, FromAten) {
 TEST(BackendDeviceTest, ToAten) {
   auto device = backendDeviceToAtenDevice(BackendDevice());
   EXPECT_EQ(device.type(), c10::kLazy);
-  EXPECT_TRUE(device.has_index());
-  EXPECT_EQ(device.index(), 0);
+  EXPECT_FALSE(device.has_index());
```

Comment:
This is due to `c10::Device` also treating `-1` has a default index/ordinal. And `c10::Device::has_index()` returns `false` if ordinal is `-1`. 

Reference: https://github.com/pytorch/pytorch/blob/master/c10/core/Device.h#L75. .

Reply:
Can we follow the convention of c10::Device and rename is_valid to has_index?

Reply:
Updated.